ApplicationInsightsInternalLogger/src/main/java/com/microsoft/applicationinsights/internal/logger/InternalLogger.java

```diff
@@ -292,6 +357,9 @@ private void onInitializationError(String errorMessage) {
     }
 
     private void log(LoggingLevel requestLevel, String message, Object... args) {
+        if (!initialized && !systemPropertyInitialize()) {
```

Comment:
It may not appear so, but this is uses double-checked locking since the method is `synchronized`

Reply:
TODO: short circuit when level==OFF